### PR TITLE
feat(behavior_path_planner): enable auto mode without rtc_auto_mode_manager

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
## Description

Enable auto mode without service from `rtc_auto_mode_manager` only with new architecture.
launcher change is this [PR](https://github.com/autowarefoundation/autoware_launch/pull/387)

TODO: 
- [x] change launch files
- [x] check behavior_path_planer_node can receive service from `rtc_auto_mode_manager` when auto_mode is enable by param

## Tests performed

checked with `lane_change_right` module with `enbale_rtc`:false and confirmed that `rtc_auto_mode_manager` can handle rtc status.

from 20 sec.
[Screencast from 2023年06月07日 14時29分31秒.webm](https://github.com/autowarefoundation/autoware.universe/assets/32741405/8a26c378-4e29-4487-b299-62cea7c8f463)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/config/scene_module_manager.param.yaml [hunk 1]
```diff
@@ -5,69 +5,79 @@
   ros__parameters:
     external_request_lane_change_left:
       enable_module: false
+      enable_auto_mode: false
```
Comment (TakaHoribe) [inline planning/behavior_path_planner/config/scene_module_manager.param.yaml:8]: The name `auto_mode` looks ambiguous for users. It should be more specific, for example, `enable_rtc`. (enable_rtc itself is not clear to understand the behavior. The description should be on the readme.)

**Review by satoshi-ota — APPROVED** [review]
Looks good to me.

File: planning/rtc_interface/include/rtc_interface/rtc_interface.hpp [hunk 2]
```diff
@@ -46,7 +46,7 @@ using unique_identifier_msgs::msg::UUID;
 class RTCInterface
 {
 public:
-  RTCInterface(rclcpp::Node * node, const std::string & name);
+  RTCInterface(rclcpp::Node * node, const std::string & name, const bool enable_rtc = false);
```
Comment (rej55) [inline planning/rtc_interface/include/rtc_interface/rtc_interface.hpp:49]:
```suggestion
  RTCInterface(rclcpp::Node * node, const std::string & name, const bool enable_rtc = true);
```

Comment (rej55) [inline planning/rtc_interface/include/rtc_interface/rtc_interface.hpp:49]:
```suggestion
  RTCInterface(rclcpp::Node * node, const std::string & name, const bool enable_rtc = true);
```

**Review by rej55 — APPROVED** [review]

**Pull request merged**